tasks/main.yml

```diff
@@ -2,18 +2,14 @@
 - name: set ip_forward to 1 to allow kickstarting of network nodes
   sysctl: name=net.ipv4.ip_forward value=1 sysctl_set=yes reload=no
 
-- name: get iptables rules
-  shell: iptables -L; iptables -t nat -L
-  register: iptablesrules
+- name: get firewalld state
+  shell: firewall-cmd --state
+  ignore_errors: true
+  always_run: yes
+  register: firewalld_state
 
-- name: allow forwarding traffic in iptables
-  command: iptables -I FORWARD 1 -i "{{ internal_interface }}" -o "{{ external_interface }}" -m comment --comment "forward from internal net" -j ACCEPT
-  when: iptablesrules.stdout.find("forward from internal net") == -1
+- include: iptables.yml
+  when: firewalld_state.rc != 0
 
-- name: allow related and established traffic in forwarding chain in iptables
-  command: iptables -I FORWARD 1 -i "{{ external_interface }}" -o "{{ internal_interface }}" -m state --state RELATED,ESTABLISHED -m comment --comment "related and established" -j ACCEPT
-  when: iptablesrules.stdout.find("related and established") == -1
-
-- name: enable masquerading
-  command: iptables -t nat -A POSTROUTING -s "{{ internal_net }}" -o "{{ external_interface }}" -m comment --comment "masquerading for internal net" -j MASQUERADE
-  when: iptablesrules.stdout.find("masquerading for internal net") == -1
+- include: firewalld.yml
+  when: firewalld_state.rc != 1
```

Comment:
If "firewall-cmd --state" were to give some other return code than 0 or 1, then both configuration options iptables and firewalld would be taken, right? That might cause a mess.